### ISSUE TITLE
The payment sheets flickers when closing iOS 18 Beta 8

### ISSUE THREAD

**Issue by aziryasin** [stripe/stripe-react-native#1723]
**Describe the bug**
When closing the payment sheet or dismissing it by pressing outside the modal, the sheet closed and pop ups again and then closes.

**To Reproduce**
1. initPaymentSheet
2. Open payment sheet using presentPaymentSheet
3. Close or press outside the sheet to dismiss
4. See the flickering

**Expected behavior**
The sheet should be closed without flickering

**Screenshots**
https://github.com/user-attachments/assets/05715492-a17c-4d0c-9fb0-4004d279bfd1

**Smartphone:**
 - Device: [e.g. iPhone6] iPhone SE 3rd Generation
 - OS: iOS 18 Beta 8
 - Version
  stripe-react-native v0.38.4
  react-native v0.70.8

**Additional context**
Only happening in iOS 18

**Comment by nicolasdevienne** [comment]
same issue when user confirm the payment

**Comment by Ortbri** [comment]
same here too

**Comment by mikkoseppa** [comment]
Same issue here. Confirmed happening on iOS 18.0, and confirmed it's still present on iOS 18.1 beta 4. Our team would really appreciate someone taking a look into it since it downgrades the customer experience of our production application, thank you!

**Comment by yuki-stripe** [comment]
Thanks for reporting this. Can you provide logs when the dismiss happens? It seems like an autolayout constraint is breaking.

**Comment by mikkoseppa** [comment]
> Thanks for reporting this. Can you provide logs when the dismiss happens? It seems like an autolayout constraint is breaking.

This is the only log entry when I dismiss. Nothing about autolayout / constraint errors.
`LOG ANALYTICS: mc_dismiss - [(key: "pay_var", value: "legacy"), (key: "ocr_type", value: "none"), (key: "apple_pay_enabled", value: 1)]`

I also checked the "Debug view hierarchy" in Xcode for any runtime constraint warnings, and there is nothing meaningful for this issue.

**Comment by yuki-stripe** [comment]
Hrmm thank you. I can't repro on iOS 18 & Xcode 16, simulator or device. We'll need to try React Native; I can't see anything in the React Native SDK that would cause this, though (eg [this code](https://github.com/stripe/stripe-react-native/blob/888c6cb31d7a00b362832e0566ac2d08890e5445/ios/StripeSdk.swift#L216))

There's an similar animation glitch that can occur on iOS 18 when a TextField takes focus when PaymentSheet dismisses _and_ the keyboard is toggled (eg "Toggle Software Keyboard" in the simulator), but that doesn't seem to be happening here.

**Comment by hot2eatorg** [comment]
Same issue here. Breaks on iOS 18. Same code works fine without the flickering on iOS 17.4 Both compiled using Xcode 16

```json
   "expo": "~50.0.14",
   "@stripe/stripe-react-native": "^0.38.6",
   "react-native": "0.73.6",
```

- Device for iOS 18: iPhone 16 Pro
- Device for iOS 17.4 iPhone 15 Pro

**Comment by v-kravchuk** [comment]
Same issue. Occurs on a real device with iOS 18

**Comment by cesarbarone** [comment]
Same issue here!

**Comment by sigaits** [comment]
I'm having the same issue on my React Native app. I've verified that it's related to iOS 18 (simulator or device). It works fine on iOS 17. Closing the payment sheet closes it, but then it pops, sweeps down the screen, and disappears. See video.

I'm using "@stripe/stripe-react-native": "^0.39.0".

https://github.com/user-attachments/assets/a6d266d4-abbc-4ecf-85e9-c232dedfe5f9

**Comment by Tieu-Mai-Nhi** [comment]
same issue with ios 18

**Comment by melyux** [comment]
Makes our app look bad.

**Comment by konstmar-ion** [comment]
same issue here with ios 18

**Comment by Ortbri** [comment]
> Hrmm thank you. I can't repro on iOS 18 & Xcode 16, simulator or device. We'll need to try React Native; I can't see anything in the React Native SDK that would cause this, though (eg [this code](https://github.com/stripe/stripe-react-native/blob/888c6cb31d7a00b362832e0566ac2d08890e5445/ios/StripeSdk.swift#L216))
> 
> There's an similar animation glitch that can occur on iOS 18 when a TextField takes focus when PaymentSheet dismisses _and_ the keyboard is toggled (eg "Toggle Software Keyboard" in the simulator), but that doesn't seem to be happening here.

Here’s the environment and the specific logs that appear when I open and close the Payment Sheet:

Environment:
React Native Version: 0.74.5
Expo SDK Version: ~51.0.37
Stripe SDK Version: @stripe/stripe-react-native: 0.37.2
Device: Physical iPhone running iOS 18.0.1 & iPhone 16 Pro Max on IOS. 18.0
Build Setup: I built the app using Xcode for my physical device to gather the logs.
Logs:
On Opening the Payment Sheet:

`
-[RemoteTextInput] -[RTIInputSystemClient
remoteTextInputSessionWithID:performInputOperation:]  perform input operation
requires a valid sessionID. inputModality = Keyboard, inputOperation = <null
selector>, customInfoType = UIEmojiSearchOperations
`



On Closing the Payment Sheet:
`
-[RemoteTextInput] -[RTIInputSystemClient
remoteTextInputSessionWithID:performInputOperation:]  perform input operation
requires a valid sessionID. inputModality = Keyboard, inputOperation = <null
selector>, customInfoType = UIEmojiSearchOperations
`
I see this error both when the sheet is opened and when it is dismissed. I haven’t seen any specific autolayout constraint warnings in the logs so far. This is the first time seeing this log besides the normal 

Not sure if this is any help, Thanks!

**Comment by erik-lissen** [comment]
same issue for us

**Comment by erik-lissen** [comment]
hi @davidme-stripe 
I can see this has been fixed in the ios SDK - I'm unfamiliar with the release processes here, how long does it usually take for the RN SDK to be updated and released with the new ios fix?

thanks a lot

**Comment by vogdev** [comment]
same issue with ios 18.0

**Comment by Ortbri** [comment]
> hi @davidme-stripe I can see this has been fixed in the ios SDK - I'm unfamiliar with the release processes here, how long does it usually take for the RN SDK to be updated and released with the new ios fix?
> 
> thanks a lot

i think we're just waiting for them to increment the version

the latest version of React native stripe being 39.0 which had 
- Updated stripe-ios to 23.30.*

The push that has the fix is on 
- stripe-ios 23.32

**Comment by Ortbri** [comment]
Hi  @charliecruzan-stripe is it possible to bump up stripe-ios ?

**Comment by fekajin** [comment]
same issue

**Comment by glopez-askit** [comment]
Same here, waiting for the update release

**Comment by dgreasi** [comment]
Same issue in iOS simulator with iOS `v18.1`

Env:
```
"react-native": "0.72.5",
"@stripe/stripe-react-native": "^0.40.0",
```

**Comment by victorious85** [comment]
The same issue in iOS simulator with iOS v18.1

**Comment by bilousov94** [comment]
The same issue

**Comment by guilean** [comment]
Same issue

**Comment by QuantGeekDev** [comment]
Same issue! :(

**Comment by igpad94** [comment]
Still having this problem with @stripe/stripe-react-native version 0.40.0, has anyone found a way to fix it?

**Comment by malthewejnold** [comment]
Same issue... Any fixes?

**Comment by alexkey89** [comment]
same issue

**Comment by justynpollard1** [comment]
same issue

**Comment by vitalyiegorov** [comment]
same issue

**Comment by alexkey89** [comment]
Seems to be fixed in latest update. 0.41.0

**Comment by Ortbri** [comment]
> Seems to be fixed in latest update. 0.41.0

Christmas update

**Comment by dgreasi** [comment]
> Same issue in iOS simulator with iOS `v18.1`, `v18.2`
> 
> Env:
> 
> ```
> "react-native": "0.72.5",
> "@stripe/stripe-react-native": "^0.40.0",
> ```

After upgrading to latest `"@stripe/stripe-react-native": "^0.41.0",` everything works as expected

**Comment by vogdev** [comment]
> @stripe/stripe-react-native": "^0.41.0"

this is working fine now 👍

**Comment by ErionTp** [comment]
Im still having this issue:

`@stripe/stripe-react-native": "^0.41.0"`

**Comment by silegrb** [comment]
> > @stripe/stripe-react-native": "^0.41.0"
> 
> this is working fine now 👍

I tried using both 0.41.0 and 0.42.0 on React Native 0.76.7, and its still flickering... Any ideas why, how did you fix it?

**Comment by melyux** [comment]
Mine did that even after I updated it. I think I had to do npm install before making a new build and then it worked.

**Comment by justynpollard1** [comment]
> Mine did that even after I updated it. I think I had to do npm install before making a new build and then it worked.

You would need to pod update/install as well, since the issue that caused it was the out of date pod file being used on the previous version

**Comment by melyux** [comment]
Yeah not `npm install`, it was actually a `npx expo prebuild --clean` that did it.

**Comment by Sorono-Dev** [comment]
on Expo 52.0.39 and stripe 0.43.0 I still got the issue

**Comment by melyux** [comment]
@Sorono-Dev Did you do a prebuild clean before building?

**Comment by chucktang786** [comment]
I am also seeing this flickering on my real device and I am on stripe 0.49.0
